types/ramda/ramda-tests.ts

```diff
@@ -2197,6 +2197,7 @@ class Rectangle {
 () => {
     const x: number[] = R.ap([R.multiply(2), R.add(3)], [1, 2, 3]); // => [2, 4, 6, 4, 5, 6]
     const y: number[] = R.ap([R.multiply(2), R.add(3)])([1, 2, 3]); // => [2, 4, 6, 4, 5, 6]
+    const z: string = R.ap<string, string, string>(R.concat, R.toUpper)('Ramda'); // => 'RamdaRAMDA'
```

Comment:
Will this actually require providing type arguments? If not, please remove them. If so, then maybe we should consider the declaration here.

Reply:
Provide the type arguments explicitly is required here.
This is cause by Typescript `strange` type infer behavior of overload function.
In this case, the `concat`  function not be infer `correctly`(you can see it be infer a very not make sense if you check the code in your editor), so we need provider type arguments
explicitly.

See [this issues](https://github.com/microsoft/TypeScript/issues/32081)